src/Microsoft.Azure.WebJobs.Host/Bindings/ClonerContext.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Description;
+using Microsoft.Azure.WebJobs.Host.Bindings.Path;
+
+namespace Microsoft.Azure.WebJobs.Host.Bindings
+{
+    /// <summary>
+    ///  Context passed to <see cref="AutoResolveAttribute"/> to aide in resolution. 
+    /// </summary>
+    internal class ClonerContext
+    {
+        public string MethodName { get; set; }
+
+        public INameResolver NameResolver { get; set; }
+
+        public static ClonerContext New(INameResolver resolver)
+        {
+            return new ClonerContext
+            {
+                NameResolver = resolver,
+            };
+        }
+
+        public static ClonerContext New(INameResolver resolver, ParameterInfo parameter)
```

Comment:
This is all internal. When we implement return values, we can adjust this as needed. (perhaps we pass in the member.NAme directly instead of ParameterInfo)

---
In reply to: [113583638](https://github.com/Azure/azure-webjobs-sdk/pull/1124#discussion_r113583638) [](ancestors = 113583638)